### PR TITLE
Make sure we prepopulate the credentials form the latest active chan…

### PR DESCRIPTION
**Pull request by norkans7**
…nel the next time we try to connect a new one

### CODE REVIEW

File: temba/channels/types/twilio/views.py [hunk 1]
```diff
@@ -426,6 +426,22 @@ def clean(self):
     menu_path = "/settings/workspace"
     title = "Connect Twilio"
 
+    def derive_initial(self):
+        initial = super().derive_initial()
+
+        org = self.request.org
+
+        last_twilio_channel = (
+            Channel.objects.filter(is_active=True, org=org, channel_type__in=["T", "TMS", "TWA"])
```
Comment (rowanseymour) [inline temba/channels/types/twilio/views.py:435]: I think as a general rule, it's preferanble to write this like `org.channels.filter(...)`

I wish we could figure out a way to not need the raw channel codes here... but it's fine for now

File: temba/channels/types/twilio/views.py [hunk 2]
```diff
@@ -426,6 +426,22 @@ def clean(self):
     menu_path = "/settings/workspace"
     title = "Connect Twilio"
 
+    def derive_initial(self):
```
Comment (rowanseymour) [inline temba/channels/types/twilio/views.py:429]: Feels a little odd to just echo the token to the user here, as on the Twilio site it's treated like a password field which you have to click Show on to see, but I guess we always treat channel tokens like that.

Maybe next step is like @ericnewcomer suggested - we skip this view altogether if we can find credentials to stuff in the session, and we let the user go back to change them, e.g. something like..

Captura de pantalla 2023-05-08 a la(s) 11 21 42

But I think that also requires checking that the token is valid - we can't let people go to the search page with credentials we haven't validated still work.

Comment (norkans7) [reply temba/channels/types/twilio/views.py:429]: Okay, I have another try on that

Comment (rowanseymour) [reply temba/channels/types/twilio/views.py:429]: Hold on - let's hear from @ericnewcomer

Comment (norkans7) [reply temba/channels/types/twilio/views.py]: I went ahead and added that in https://github.com/nyaruka/rapidpro/pull/4536/commits/0b5189e290ddfa4e6444a572f3220ea447914c80

Comment (norkans7) [reply temba/channels/types/twilio/views.py]: so a reset link with not use initials on the connect form

Comment (norkans7) [reply temba/channels/types/twilio/views.py]: Monosnap Connect Twilio 2023-05-08 19-53-38

File: temba/channels/types/twilio/views.py [hunk 3]
```diff
@@ -426,6 +428,26 @@ def clean(self):
     menu_path = "/settings/workspace"
     title = "Connect Twilio"
 
+    def pre_process(self, *args, **kwargs):
+        reset_creds = self.request.GET.get("reset_creds", "")
+        org = self.request.org
+
+        last_twilio_channel = (
+            org.channels.filter(is_active=True, channel_type__in=["T", "TMS", "TWA"]).order_by("-created_on").first()
```
Comment (norkans7) [inline temba/channels/types/twilio/views.py:436]: @rowanseymour is it better to use method on the channel type, and we get the latest credentials from that method?

`def get_latest_credentials(self, org)`

we use that in the views using `self.channel_type.get_latest_credentials(self.request.org)`

File: temba/channels/types/twilio/views.py (hunk 2 above)
Comment (ericnewcomer) [reply temba/channels/types/twilio/views.py]: This is great. I like it a lot.

**Pull request merged**